### ISSUE TITLE
segmentation fault

### ISSUE THREAD

**Issue by henrikrudstrom** [Cutehacks/gel#4]
i get a segmentation fault when i set the collection comparator to a function or a string
the stack trace points to:
JsonListModel::extractRoles line 20
JsonListModel::add  line103

this is my model: 

```
JsonListModel {
        id: jsonModel
        Component.onCompleted: {
            randomIds(24)
            var date = Date.now()
            for(var i = 0; i < 100; i++){
                var tx = randomTransaction(date)
                add(tx)
            }
        }
    }
```

```
    Collection{
        id: collection
        model: TransactionStore.model
        comparator: "date"
        filter: function(item) {
            console.log("filter", item.amount)
            return item.amount < 100
        }
    }
```

Im running qt 5.5 on ubuntu 14, didnt use qpm to install, just submoduled and included the .pri

Any idea

**Comment by jrbarron** [comment]
@henrikrudstrom Are you using the latest version? There's been a few fixes in the last few days. In any case, I'll take a look.

**Comment by henrikrudstrom** [comment]
I cloned it yesterday. noticed now that it doesnt crash on my windows computer. 
also noticed that i refered to `collection.model` and not just `collection` in my listview, not sure if that makes the difference, will check when i get home.

**Comment by jrbarron** [comment]
Ok, we just pushed another fix around the roles stuff so be sure to pull :)

**Comment by henrikrudstrom** [comment]
thanks for the quick response, will check when i get home.

**Comment by henrikrudstrom** [comment]
Still same problem on linux. I should mention the json list is instantiated in a singleton. 
Also noticed some peculiar behaviour, if i add a console.log before adding to a transaction it works fine.

´´´

```
property JsonListModel model: JsonListModel {
    Component.onCompleted: {

        randomIds(24)
        var date = Date.now()
        for(var i = 0; i < 100; i++){
            var tx = randomTransaction(date)
            //logging the tx apparently does the trick
            console.log(tx)
            date = tx.date
            add(tx)
        }
    }
}
```

´´´

**Comment by capisce** [comment]
The latest version should have a fix for this crash now (Qt 5.6 also has a fix for the crash, but for earlier Qt versions we need to work around it)

**Comment by capisce** [comment]
The latest version should have a fix for this crash (there's a fix in Qt 5.6 too, but for earlier versions we need to work around it)

**Comment by jrbarron** [comment]
Closing this for now since it should be fixed.
